extensions/common/transaction/transaction-local/src/main/java/org/eclipse/edc/transaction/local/LocalTransactionContext.java

```diff
@@ -100,13 +113,25 @@ public void registerResource(LocalTransactionResource resource) {
 
     private static class Transaction {
         private boolean rollbackOnly = false;
+        private List<TransactionSynchronization> synchronizations;
```

Comment:
this could be initialized with a new `ArrayList`, so the null checks at line 127 and 131 will disappear

Reply:
Yeah, I did this on purpose as a performance optimization. For transactions that don't have synchronization, we can avoid the overhead of an ArrayList creation.